packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -338,25 +345,24 @@ export default defineComponent({
     },
 
     isPasswordEnforcedFor(link) {
-      const currentRole = LinkShareRoles.getByBitmask(
-        link.permissions,
-        link.indirect || this.resource.isFolder
-      )
-
-      const canRead = currentRole.hasPermission(SharePermissions.read)
-      const canUpdate = currentRole.hasPermission(SharePermissions.update)
-      const canCreate = currentRole.hasPermission(SharePermissions.create)
-      const canDelete = currentRole.hasPermission(SharePermissions.delete)
-
-      const isReadOnly = canRead && !canUpdate && !canCreate && !canDelete
-      const isUploadOnly = !canRead && !canUpdate && canCreate && !canDelete
-      const isReadWrite = canRead && !canUpdate && canCreate && !canDelete
-      const isReadWriteDelete = canRead && canUpdate && canCreate && canDelete
+      const isFolder = link.indirect || this.resource.isFolder
+      const currentRole = LinkShareRoles.getByBitmask(link.permissions, isFolder)
+
+      /**
+       * `passwordEnforced` members are oddly designed. they look like they map to permissions,
+       * but in reality they map to role names. hence the comparison with specific link roles.
```

Comment:
wanted to add that I feel betrayed by the capability. It looks so promising regarding permissions... but it's a big nope in reality.